File: src/utils/validate_birth_date.py
from datetime import datetime
from messages import messages


def validate_birth_date(d):
    try:
        birth_date = datetime.fromisoformat(d)
        current_date = datetime.now()

        if birth_date > current_date:
            raise ValueError(messages["bad_birth_date"])

    except ValueError:
        raise ValueError(messages["bad_birth_date"])

File: src/utils/validate_birth_date.dart
import 'messages.dart';

void validateBirthDate(String d) {
  try {
    DateTime date = DateTime.parse(d);

    if (date.isAfter(DateTime.now())) {
      throw Exception(messages['bad_birth_date']);
    }
  } catch (_) {
    throw Exception(messages['bad_birth_date']);
  }
}
